elfpy/eth/transactions.py

```diff
@@ -62,6 +64,89 @@ def smart_contract_read(contract: Contract, function_name: str, *fn_args, **fn_k
     return {f"value{idx}": value for idx, value in enumerate(return_values)}
 
 
+async def async_wait_for_transaction_receipt(
+    web3: Web3, transaction_hash: HexBytes, timeout: float = 120, poll_latency: float = 0.1
+) -> TxReceipt:
+    """Async version of wait_for_transaction_receipt
+    This function is copied from `web3.eth.wait_for_transaction_receipt`, but using a non-blocking wait
+    instead of a blocking wait
+    """
+    try:
+        with Timeout(timeout) as _timeout:
+            while True:
+                try:
+                    tx_receipt = web3.eth.get_transaction_receipt(transaction_hash)
+                except TransactionNotFound:
+                    tx_receipt = None
+                if tx_receipt is not None:
+                    break
+                await _timeout.async_sleep(poll_latency)
+        return tx_receipt
+
+    except Timeout:
+        # pylint: disable=raise-missing-from
```

Comment:
adding in the `from` is actually pretty nice, instead of disabling. It's also easy. Here's a solution:
https://realpython.com/python-raise-exception/#:~:text=You'll%20typically%20use%20from,__cause__%20attribute.